### ISSUE TITLE
Custom game theme fails to load on Linux

### ISSUE THREAD

**Issue by bluebarnacles** [godotengine/godot#52486]
### Godot version

v3.4.beta5.official [dd0ee4872], v3.4.beta4.official [6a058cbf3], v3.4.beta3.official [8db0bd442]

### System information

Linux Mint 20.2 Uma (based on Ubuntu 20.04)

### Issue description

I can load a theme directly on a Control in Godot v3.4.beta4.official (and beta3), but if I use the same theme file as the game's custom theme, the theme fails to load, with this error message:

```
ERROR: Condition "img.is_null() || img->empty()" is true. Returned: ERR_FILE_CORRUPT
   at: _load_data (scene/resources/texture.cpp:566)
ERROR: Failed loading resource: res://.import/button.png-234620e182281afdeb4aab4d2ed4f8a7.stex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:270)
ERROR: Failed loading resource: res://button.png. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:270)
ERROR: res://theme.tres:3 - Parse Error: [ext_resource] referenced nonexistent resource at: res://button.png
   at: poll (scene/resources/resource_format_text.cpp:412)
ERROR: Failed to load resource 'res://theme.tres'.
   at: load (core/io/resource_loader.cpp:206)
ERROR: Failed loading resource: res://theme.tres. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:270)
ERROR: Error loading custom theme 'res://theme.tres'
   at: register_scene_types (scene/register_scene_types.cpp:793)
```

When loading the bad project in Godot v3.4.beta2.official, the editor crashes immediately with this message:

```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7f78f19b9210] (??:0)
-- END OF BACKTRACE --
```

### Steps to reproduce

Add a custom theme that loads a PNG resource. Try to open that project on Linux.

### Minimal reproduction project

The project that fails to load:
[custom_game_theme.zip](https://github.com/godotengine/godot/files/7127369/custom_game_theme.zip)

The working project:
[manual_theme.zip](https://github.com/godotengine/godot/files/7127365/manual_theme.zip)

**Comment by Calinou** [comment]
Does it work in 3.3.3?

**Comment by bluebarnacles** [comment]
No, 3.3.3 gives a similar error to 3.4beta1 and beta2 on load of the project with custom theme:

```
Godot Engine v3.3.3.stable.official.b973f997f - https://godotengine.org
OpenGL ES 2.0 Renderer: Mesa DRI Intel(R) HD Graphics 5000 (HSW GT3)
OpenGL ES Batching: ON
 
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7fbe8e721210] (??:0)
-- END OF BACKTRACE --
```

3.3.3 can load the project that has the theme applied on a Control.

**Comment by dalexeev** [comment]
Possibly related: #52262.

**Comment by autofool** [comment]
Maybe similar issue, I'm not sure but using custom theme file in theme editor makes Godot not launch any projects. It just crashes.

**Comment by DaSlang** [comment]
The exact same thing is happening on Godot v3.4.stable.official `206ba70f4` (on MacOS Big Sur 11.6.2).

A custom game theme fails to load and shows the same error message:
```
E 0:00:00.161   _load_data: Condition "img.is_null() || img->empty()" is true. Returned: ERR_FILE_CORRUPT
  <C++ Source>  scene/resources/texture.cpp:566 @ _load_data()

E 0:00:00.161   _load: Failed loading resource: res://.import/box.jpg-dd5cdbbb4a401123f845c68461a35987.stex. Make sure resources have been imported by opening the project in the editor at least once.
  <C++ Error>   Condition "found" is true. Returned: RES()
  <C++ Source>  core/io/resource_loader.cpp:271 @ _load()

E 0:00:00.161   _load: Failed loading resource: res://assets/images/box.jpg. Make sure resources have been imported by opening the project in the editor at least once.
  <C++ Error>   Condition "found" is true. Returned: RES()
  <C++ Source>  core/io/resource_loader.cpp:271 @ _load()

E 0:00:00.161   poll: res://assets/my_theme.tres:4 - Parse Error: [ext_resource] referenced nonexistent resource at: res://assets/images/box.jpg
  <C++ Source>  scene/resources/resource_format_text.cpp:412 @ poll()

E 0:00:00.161   load: Failed to load resource 'res://assets/my_theme.tres'.
  <C++ Error>   Condition "err != OK" is true. Returned: RES()
  <C++ Source>  core/io/resource_loader.cpp:206 @ load()

E 0:00:00.161   _load: Failed loading resource: res://assets/my_theme.tres. Make sure resources have been imported by opening the project in the editor at least once.
  <C++ Error>   Condition "found" is true. Returned: RES()
  <C++ Source>  core/io/resource_loader.cpp:271 @ _load()

E 0:00:00.161   register_scene_types: Error loading custom theme 'res://assets/my_theme.tres'
  <C++ Source>  scene/register_scene_types.cpp:795 @ register_scene_types()

```

**Comment by akien-mga** [comment]
This was fixed in 3.4.1: https://godotengine.org/article/maintenance-release-godot-3-4-1

(And 3.4.2 is available too with more fixes.)
